app/controllers/application_controller.rb

```diff
@@ -860,19 +860,12 @@ def get_reports_menu(group = current_group, tree_type = "reports", mode = "menu"
     folders = []
     user = current_user
     @sb[:grp_title] = reports_group_title
-    @data = []
+    data = []
     if !group.settings || group.settings[:report_menus].blank? || mode == "default"
       # array of all reports if menu not configured
-      @rep = MiqReport.for_user(current_user).sort_by { |r| [r.rpt_type, r.filename.to_s, r.name] }
-      if tree_type == "timeline"
-        @data = @rep.reject { |r| r.timeline.nil? }
-      else
-        @data = @rep.select do |r|
-          r.template_type == "report" && !r.template_type.blank?
-        end
-      end
-      @data.each do |r|
-        next if r.template_type != "report" && !r.template_type.blank?
+      data = MiqReport.for_user(current_user).where(:template_type => "report").order(:rpt_type, :filename, :name)
```

Comment:
Aha, found it.

PostgreSQL is using `en_US.UTF-8` collation in it's sort.  Ruby is using `POSIX` (or probably some system default, which for me is likely posix).

If I change my PostgreSQL query to use posix collation, I get the same results as the Ruby in-memory sort.

I'm sure there's probably no way to remedy this on an actual appliance.  So, I guess it comes down to whether it's worth allowing the DB to do the sort and getting a slightly different sort order, or better to keep the same sort order and do the sort in memory.